### PR TITLE
Tmfset floating menu events

### PR DESCRIPTION
**Pull request by odevices**

### CODE REVIEW

**Comment by odevices** [comment]
@tmfset Did you get notified when I created this?

**Comment by tmfset** [comment]
Ha nope I'd need an @tmfset

**Comment by odevices** [comment]
I didn't really touch the stuff that you changed.  I just split  Overlay.lua into 2 child modules, Overlay.Message and Overlay.FloatingMenu/  Then I took advantage of the new organization to simplify the naming in some places.

File: xroot/Settings/init.lua [hunk 1]
```diff
@@ -274,7 +274,18 @@ local defaults = {
...
+  },
+  floatingMenuDelay = {
+    category = "General",
+    description = "Long press time for floating menu:",
+    value = "0.3s",
+    choices = {
+      "0.1s",
+      "0.2s",
+      "0.3s",
+      "0.4s"
```
Comment (tmfset) [inline xroot/Settings/init.lua:286]: I _suspect_ our time sense is somewhat exponential. Consider 0.05s over 0.4s.

**Review by tmfset — APPROVED** [review]
Looks good to me

**Comment by odevices** [comment]
Here is a test firmware.  If you could give it a whirl against your client code I would be grateful:

[er-301-v0.6.16-testing.zip](https://github.com/odevices/er-301/files/7109159/er-301-v0.6.16-testing.zip)

File: xroot/Overlay/FloatingMenu.lua [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+end
+
+local function getDelayTime()
+  local Settings = require "Settings"
+  local choice = Settings.get("floatingMenuDelay")
+  local values = {
+    ["0.1s"] = 0.1,
+    ["0.2s"] = 0.2,
+    ["0.3s"] = 0.3,
+    ["0.5s"] = 0.5,
```
Comment (tmfset) [inline xroot/Overlay/FloatingMenu.lua:154]: Ha well I meant 50ms instead of 500ms :p

Comment (odevices) [reply xroot/Overlay/FloatingMenu.lua:154]: Oops

Comment (odevices) [reply xroot/Overlay/FloatingMenu.lua]: Now 50ms, 0.1s, 0.2s, 0.4s.

50ms is really short for a long press! :thinking: 

[er-301-v0.6.16-testing.zip](https://github.com/odevices/er-301/files/7109165/er-301-v0.6.16-testing.zip)

Comment (tmfset) [reply xroot/Overlay/FloatingMenu.lua]: Ha yea I'm running it now and 0.1s is also proving to be _quite_ short.. trying the others now

Comment (tmfset) [reply xroot/Overlay/FloatingMenu.lua]: Ok the difference between 50 and 100ms is basically not noticeable, how about this:

100ms - Very short, practically instant
300ms - Normal, almost always intentional
500ms - Slow, extremely intentional

That's my take away at least. I'm realizing that 0.3s was probably the best choice in the first place so.. my bad ha.

Comment (odevices) [reply xroot/Overlay/FloatingMenu.lua]: That matches my experience also.

**Comment by odevices** [comment]
Alright I'm going to merge to master unless you have any objections?

**Review by tmfset — APPROVED** [review]
No more nits :) Looks good to me

**Comment by odevices** [comment]
Thanks for the great collaboration!

**Pull request merged**